crates/assistant/src/assistant_settings.rs

```diff
@@ -201,42 +202,26 @@ impl AssistantSettingsContent {
             AssistantSettingsContent::Versioned(settings) => match settings {
                 VersionedAssistantSettingsContent::V1(settings) => settings.clone(),
             },
-            AssistantSettingsContent::Legacy(settings) => {
-                if let Some(open_ai_api_url) = settings.openai_api_url.as_ref() {
-                    AssistantSettingsContentV1 {
-                        button: settings.button,
-                        dock: settings.dock,
-                        default_width: settings.default_width,
-                        default_height: settings.default_height,
-                        provider: Some(AssistantProvider::OpenAi {
-                            default_model: settings
-                                .default_open_ai_model
-                                .clone()
-                                .unwrap_or_default(),
-                            api_url: open_ai_api_url.clone(),
-                        }),
-                    }
+            AssistantSettingsContent::Legacy(settings) => AssistantSettingsContentV1 {
+                enabled: None,
+                button: settings.button,
+                dock: settings.dock,
+                default_width: settings.default_width,
+                default_height: settings.default_height,
+                provider: if let Some(open_ai_api_url) = settings.openai_api_url.as_ref() {
```

Comment:
This part of the diff it just a refactoring, as I noticed that the branching only applied to the `provider` field.